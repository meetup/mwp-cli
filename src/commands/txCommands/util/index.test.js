const txlib = require('./index');

describe('trn utils', () => {
	it('compares two objects, returning an object composed of new keys or existing keys with changed msgstr values', () => {
		const same = { msgstr: 'same' };
		const different = { msgstr: 'abc' };
		const newEntry = { msgstr: 'new' };

		const mock = [
			{
				same,
				different: { msgstr: '123' },
			},
			{
				same,
				different,
				newEntry,
			},
		];

		const expected = {
			different,
			newEntry,
		};

		expect(txlib.diff(mock)).toEqual(expected);
	});

	it('wraps Po objects', () => {
		const trnObjs = [{}, {}];
		expect(txlib.wrapPoTrns(trnObjs)).toMatchSnapshot();
	});

	it('convert react-intl style objects into po style objects', () => {
		const trnObjs = [
			{
				id: 'id1',
				defaultMessage: 'id1 text',
				description: { text: 'text for test', jira: 'MW-001' },
				file: 'text.txt',
				start: { line: 2, column: 5 },
			},
			{
				id: 'id2',
				defaultMessage: 'id2 text',
				description: { text: 'text for test v2', jira: 'MW-002' },
				file: 'text2.txt',
				start: { line: 10, column: 10 },
			},
		];

		expect(txlib.reactIntlToPo(trnObjs)).toMatchSnapshot();
	});

	it('merge objects, throw error if dupe key', () => {
		const msg1 = {
			id1: {
				msgid: 'id1',
				msgstr: ['id1 text'],
				comments: {
					extracted: 'text for test',
					reference: 'text.txt:2:5',
				},
			},
		};
		const msg2 = {
			id2: {
				msgid: 'id2',
				msgstr: ['id2 text'],
				comments: {
					extracted: 'text for test v2',
					reference: 'text2.txt:10:10',
				},
			},
		};

		// expected success
		expect(txlib.mergeLocalTrns([msg1, msg2])).toMatchSnapshot();
		// expected error from duplicate msgs
		expect(() => txlib.mergeLocalTrns([msg1, msg1])).toThrow();
	});
});
