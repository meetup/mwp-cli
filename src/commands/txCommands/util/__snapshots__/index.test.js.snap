// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trn utils convert react-intl style objects into po style objects 1`] = `
Object {
  "id1": Object {
    "comments": Object {
      "extracted": "text for test",
      "reference": "text.txt:2:5",
      "translator": "MW-001",
    },
    "msgid": "id1",
    "msgstr": Array [
      "id1 text",
    ],
  },
  "id2": Object {
    "comments": Object {
      "extracted": "text for test v2",
      "reference": "text2.txt:10:10",
      "translator": "MW-002",
    },
    "msgid": "id2",
    "msgstr": Array [
      "id2 text",
    ],
  },
}
`;

exports[`trn utils loads resource list and sorts by date modified 1`] = `
Array [
  "MW-1168_gh_delete_discussion",
  "MW-1454_memberSample",
]
`;

exports[`trn utils merge objects, throw error if dupe key 1`] = `
Object {
  "id1": Object {
    "comments": Object {
      "extracted": "text for test",
      "reference": "text.txt:2:5",
    },
    "msgid": "id1",
    "msgstr": Array [
      "id1 text",
    ],
  },
  "id2": Object {
    "comments": Object {
      "extracted": "text for test v2",
      "reference": "text2.txt:10:10",
    },
    "msgid": "id2",
    "msgstr": Array [
      "id2 text",
    ],
  },
}
`;

exports[`trn utils takes po file content, extract trn content 1`] = `
Object {
  "event.oneMemberWent": Object {
    "comments": Object {
      "reference": "src/components/EventCard.jsx:27:16",
      "translator": "MW-000",
    },
    "msgid": "event.oneMemberWent",
    "msgstr": Array [
      "1 Mitglied ging",
    ],
  },
}
`;

exports[`trn utils takes trn content, returns po file 1`] = `
"msgid \\"\\"
msgstr \\"Content-Type: text/plain; charset=utf-8\\\\n\\"

# MW-000
#: src/components/EventCard.jsx:27:16
msgid \\"event.oneMemberWent\\"
msgstr \\"1 Mitglied ging\\"
"
`;

exports[`trn utils wraps Po objects 1`] = `
Object {
  "charset": "utf-8",
  "headers": Object {
    "content-type": "text/plain; charset=utf-8",
  },
  "translations": Object {
    "": Array [
      Object {},
      Object {},
    ],
  },
}
`;
